src/app/showcase/app.component.ts

```diff
@@ -60,29 +94,68 @@ export class ExampleEditorOptions {
         name: 'Framework',
         description: 'Whether to include any additional framework',
         type: 'options',
-        options: ['None', {text: 'AngularJS', value: 'angular'}, {text: 'Backbone.js', value: 'backbone'}]
+        options: ['None', { text: 'AngularJS', value: 'angular' }, { text: 'Backbone.js', value: 'backbone' }]
     })
     framework = 'None';
+}
 
+export interface IExampleStudentOptions {
+    birth: string;
+    name: string;
+    description: string;
+    age: number;
+    telephone: string;
+    gender: string;
+    editor: ExampleEditorOptions;
 }
 
-export class ExampleStudentOptions {
 
-    @meta({name: 'Birth', group: 'Basic Information', type: 'date', order: 4})
+export class ExampleStudentOptions implements IExampleStudentOptions{
+
+    @meta({ name: 'Birth', group: 'Basic Information', type: 'date', order: 4 })
     birth = '2018-05-08';
 
-    @meta({name: 'Name', group: 'Basic Information', type: 'text', order: 1, link: 'http://www.baidu.com'})
+    @meta({ name: 'Name', group: 'Basic Information', type: 'text', order: 1, link: 'http://www.baidu.com' })
     name = 'Lily';
 
-    @meta({name: 'Age', group: 'Basic Information1', valueConvert: parseInt, type: 'text', order: 2})
+    @meta({ name: 'Description', group: 'Basic Information', type: 'text', readonly: true, order: 1, link: 'http://www.baidu.com' })
```

Comment:
The  `readonly`  should be  declared  here https://github.com/mokeyish/ngx-property-grid/blob/e92c040bb37f2a5bd44dd3ebd225207d48898331/projects/ngx-property-grid/src/lib/property-item-meta.ts#L5.

Reply:
Sorry, I'm not overly familiar with Git, I thought the pull request was for a commit, not for my whole branch. So I made 2 sets of changes on my fork.   

1. The 'readonly' property, which seems to work OK, but on balance I think I'd have called it 'disabled' as this would have fit in better with the Mat controls and the way your pushing properties onto dynamically created controls.
2. The second check in I made allowed you to dynamically configure the meta data. I'd not intended this to be pushed onto your branch until I'd done a fair bit more testing on it.

So I'm not sure if you can see both changes or just the 'readonly' property change?

With regards to the 'readonly' property, it is declared on PropertyItemMeta, see
https://github.com/mokeyish/ngx-property-grid/pull/35/commits/a40a2b8988d4d75d64d8841e14d4c1bc12a09578

Reply:
Ok, I have seen the `readonly` property in the `PropertyItemMeta`. What about changing `readonly` to `editable`?

Reply:
Calling it `disabled` would be better then you don't need the explicit set in https://github.com/mokeyish/ngx-property-grid/pull/35/commits/a40a2b8988d4d75d64d8841e14d4c1bc12a09578#diff-f12aa8bfa90098e759f0a7ddc4569145 it would just be applied implicitly in `Object.assign(component.instance, this.meta.options);`